sorrydb/agents/json_agent.py

```diff
@@ -0,0 +1,137 @@
+from typing import Protocol, List, Dict
+from pathlib import Path
+from sorrydb.database.sorry import Sorry, sorry_object_hook, SorryJSONEncoder
+from tempfile import TemporaryDirectory
+import json
+import logging
+import contextlib
+
+from sorrydb.database.process_sorries import build_lean_project
+from sorrydb.utils.git_ops import prepare_repository
+from sorrydb.utils.lean_repo import build_lean_project
+from sorrydb.utils.verify import verify_proof
+
+# Create a module-level logger
+logger = logging.getLogger(__name__)
+
+
+def load_sorry_json(json_path: Path) -> List[Sorry]:
+    """Load a sorry JSON file.
+
+    Args:
+        json_path: Path to the sorry JSON file
+
+    Returns:
+        List of sorries
+
+    Raises:
+        FileNotFoundError: If the JSON file doesn't exist
+        json.JSONDecodeError: If the JSON file is invalid
+    """
+    logger.info(f"Loading sorry JSON from {json_path}")
+    try:
+        with open(json_path, "r") as f:
+            sorry_data = json.load(f, object_hook=sorry_object_hook)
+
+        return sorry_data
```

Comment:
I think we want to return this to align with the return type:

```suggestion
        return sorry_data["sorries"]
```

Reply:
Indeed!